kernel/src/device/tdxguest/mod.rs

```diff
@@ -175,36 +181,34 @@ fn handle_get_report(arg: usize) -> Result<i32> {
     let report = tdx_get_report(&user_request.report_data)?;
 
     let tdx_report_vaddr = arg + TDX_REPORTDATA_LEN;
-    user_space.write_bytes(tdx_report_vaddr, &mut VmReader::from(report.as_ref()))?;
+    user_space.write_bytes(tdx_report_vaddr, &mut report.reader().limit(TDX_REPORT_LEN))?;
     Ok(0)
 }
 
-fn tdx_get_report(inblob: &[u8]) -> Result<Box<[u8]>> {
+const TDREPORT_REPORTDATA_OFFSET: usize = 128;
+
+fn tdx_get_report(inblob: &[u8]) -> Result<USegment> {
```

Comment:
Changing the return type from `Box<[u8]>` to `USegment` makes less clear the length of the returned report. Clarify this point by adding some Rust doc.

```rust
/// Gets the TDX report given the specified data in `inblob`.
///
/// The first `TDX_REPORT_LEN` bytes of data in the returned `USegment` is the report.
/// The rest in `USegment` should be ignored.
fn tdx_get_report(inblob: &[u8]) -> Result<USegment> {
```